modules/openideal_statistics/src/Plugin/Block/OpenidealStatisticsWorkflowBlock.php

```diff
@@ -86,23 +87,172 @@ public static function create(ContainerInterface $container, array $configuratio
    */
   public function build() {
     $build = [];
-    if ($node = $this->getEntity($this->getContexts())) {
-      $state = $this->moderationInformation->getOriginalState($node);
+    if (!($node = $this->getEntity($this->getContexts()))) {
+      return $build;
+    }
 
+    $state = $this->moderationInformation->getOriginalState($node);
+    if (!$this->configuration['show_all_states']) {
       $build = [
-        'status' => [
-          '#type' => 'html_tag',
-          '#tag' => 'div',
-          '#attributes' => ['class' => ['idea-statistics-and-status-block--status']],
-          '#value' => $state->label(),
+        'container' => [
+          '#type' => 'container',
+          '#attributes' => ['class' => ['idea-statistics-and-status-block--status', $state->id()]],
+          'status' => [
+            '#type' => 'html_tag',
+            '#attributes' => ['class' => ['idea-statistics-and-status-block--status__container']],
+            '#tag' => 'div',
+            '#value' => $state->label(),
+          ],
         ],
         '#cache' => [
           'tags' => $node->getCacheTags(),
         ],
       ];
     }
+    else {
+      /** @var \Drupal\content_moderation\Plugin\WorkflowType\ContentModeration $contentModerationPlugin */
+      $contentModerationPlugin = $this->entityManager->getStorage('workflow')->load('life_cycle_phases')->getTypePlugin();
+      // Hardcoded states groups FE configuration.
+      $configurations = [
+        [
+          'states' => ['draft', 'draft_approval'],
+          'label' => $this->t('prepare'),
```

Comment:
Shouldn't labels start with uppercase letter? (i.e. "Prepare" instead of "prepare")

Reply:
I transform them into Capitalized via CSS. The reason why I'm doing this that to make CSS style names be in the same case.

Reply:
I reworked logic a little bit, please check it.